include/coap3/coap_session_internal.h

```diff
@@ -313,6 +319,89 @@ coap_session_t *coap_session_new_dtls_session(coap_session_t *session,
 void coap_session_free(coap_session_t *session);
 void coap_session_mfree(coap_session_t *session);
 
+#define COAP_SESSION_REF(s) s->ref
```

Comment:
My gut feeling is that we are better of if the values defined here and in the following lines are enclosed in parentheses.

Reply:
Agreed - corrected.

Reply:
Hm, I still see these in commit 53299b3:

```
#define COAP_SESSION_REF(s) (s)->ref

/* RFC7252 */
#define COAP_ACK_TIMEOUT(s) (s)->ack_timeout
#define COAP_ACK_RANDOM_FACTOR(s) (s)->ack_random_factor
#define COAP_MAX_RETRANSMIT(s) (s)->max_retransmit
#define COAP_NSTART(s) (s)->nstart
#define COAP_DEFAULT_LEISURE(s) (s)->default_leisure
#define COAP_PROBING_RATE(s) (s)->probing_rate
```

Reply:
Do you mean `#define COAP_SESSION_REF(s) ((s)->ref)` ?

Reply:
Yes, exactly (as well as for the following defines noted).

Reply:
Updated and code pushed

Reply:
Thanks!